2025-09-26 16:00:27,655 - __main__ - INFO - üöÄ Starting Real-Time HDFS Anomaly Detection Engine...
2025-09-26 16:00:27,657 - __main__ - INFO -  Database initialized: anomaly_detection.db
2025-09-26 16:00:27,662 - __main__ - INFO - ‚úÖ Connected to Redis at localhost:6379
2025-09-26 16:00:27,701 - httpx - INFO - HTTP Request: GET http://localhost:6333 "HTTP/1.1 200 OK"
2025-09-26 16:00:27,701 - __main__ - INFO - ‚úÖ Connected to Qdrant at localhost:6333
2025-09-26 16:00:27,701 - __main__ - INFO - üì• Loading ensemble model from ./ensemble/models/line_level_ensemble_v2/line_level_ensemble_v2_results.joblib
2025-09-26 16:00:28,342 - __main__ - INFO - ‚úÖ Loaded ensemble with 3 models
2025-09-26 16:00:28,342 - __main__ - INFO - üéØ Model weights for weighted voting:
2025-09-26 16:00:28,342 - __main__ - INFO -    dt: 0.0790
2025-09-26 16:00:28,342 - __main__ - INFO -    mlp: 0.4315
2025-09-26 16:00:28,342 - __main__ - INFO -    sgd: 0.0747
2025-09-26 16:00:28,342 - __main__ - INFO -    qdrant_similarity: 0.4148
2025-09-26 16:00:28,342 - __main__ - INFO - üì° Enhanced Kafka consumer started
2025-09-26 16:00:28,343 - __main__ - INFO - ‚úÖ Real-Time Anomaly Detection Engine ready!
2025-09-26 16:00:28,343 - kafka.conn - INFO - <BrokerConnection client_id=kafka-python-2.2.15, node_id=bootstrap-0 host=localhost:9092 <connecting> [IPv6 ('::1', 9092, 0, 0)]>: connecting to localhost:9092 [('::1', 9092, 0, 0) IPv6]
2025-09-26 16:00:28,358 - kafka.conn - INFO - <BrokerConnection client_id=kafka-python-2.2.15, node_id=bootstrap-0 host=localhost:9092 <checking_api_versions_recv> [IPv6 ('::1', 9092, 0, 0)]>: Broker version identified as 2.6
2025-09-26 16:00:28,358 - kafka.conn - INFO - <BrokerConnection client_id=kafka-python-2.2.15, node_id=bootstrap-0 host=localhost:9092 <connected> [IPv6 ('::1', 9092, 0, 0)]>: Connection complete.
2025-09-26 16:00:28,359 - kafka.consumer.subscription_state - INFO - Updating subscribed topics to: ('logs',)
2025-09-26 16:00:28,359 - __main__ - INFO - üì° Enhanced Kafka consumer listening to topic: logs
2025-09-26 16:00:28,363 - kafka.conn - INFO - <BrokerConnection client_id=kafka-python-2.2.15, node_id=1 host=localhost:9092 <connecting> [IPv6 ('::1', 9092, 0, 0)]>: connecting to localhost:9092 [('::1', 9092, 0, 0) IPv6]
2025-09-26 16:00:28,467 - kafka.conn - INFO - <BrokerConnection client_id=kafka-python-2.2.15, node_id=1 host=localhost:9092 <connected> [IPv6 ('::1', 9092, 0, 0)]>: Connection complete.
2025-09-26 16:00:28,467 - kafka.conn - INFO - <BrokerConnection client_id=kafka-python-2.2.15, node_id=bootstrap-0 host=localhost:9092 <connected> [IPv6 ('::1', 9092, 0, 0)]>: Closing connection. 
2025-09-26 16:00:28,573 - kafka.cluster - INFO - Coordinator for group/anomaly_engine_1758888028 is BrokerMetadata(nodeId='coordinator-1', host='localhost', port=9092, rack=None)
2025-09-26 16:00:28,573 - kafka.coordinator - INFO - Discovered coordinator coordinator-1 for group anomaly_engine_1758888028
2025-09-26 16:00:28,574 - kafka.coordinator.heartbeat - INFO - Starting new heartbeat thread
2025-09-26 16:00:28,575 - kafka.coordinator.consumer - INFO - Revoking previously assigned partitions set() for group anomaly_engine_1758888028
2025-09-26 16:00:28,575 - kafka.coordinator - INFO - Failed to join group anomaly_engine_1758888028: NodeNotReadyError: coordinator-1
2025-09-26 16:00:28,577 - kafka.conn - INFO - <BrokerConnection client_id=kafka-python-2.2.15, node_id=coordinator-1 host=localhost:9092 <connecting> [IPv6 ('::1', 9092, 0, 0)]>: connecting to localhost:9092 [('::1', 9092, 0, 0) IPv6]
2025-09-26 16:00:28,682 - kafka.coordinator - INFO - Failed to join group anomaly_engine_1758888028: NodeNotReadyError: coordinator-1
2025-09-26 16:00:28,683 - kafka.conn - INFO - <BrokerConnection client_id=kafka-python-2.2.15, node_id=coordinator-1 host=localhost:9092 <connected> [IPv6 ('::1', 9092, 0, 0)]>: Connection complete.
2025-09-26 16:00:28,788 - kafka.coordinator - INFO - (Re-)joining group anomaly_engine_1758888028
2025-09-26 16:00:28,803 - kafka.coordinator - INFO - Received member id kafka-python-2.2.15-6c4f2f78-9515-48ad-95d9-0bdf7498d9f3 for group anomaly_engine_1758888028; will retry join-group
2025-09-26 16:00:28,803 - kafka.coordinator - INFO - Failed to join group anomaly_engine_1758888028: [Error 79] MemberIdRequiredError
2025-09-26 16:00:28,804 - kafka.coordinator - INFO - (Re-)joining group anomaly_engine_1758888028
2025-09-26 16:00:28,830 - kafka.coordinator - INFO - Successfully joined group anomaly_engine_1758888028 <Generation 1 (member_id: kafka-python-2.2.15-6c4f2f78-9515-48ad-95d9-0bdf7498d9f3, protocol: range)>
2025-09-26 16:00:28,830 - kafka.coordinator - INFO - Elected group leader -- performing partition assignments using range
2025-09-26 16:00:28,863 - kafka.consumer.subscription_state - INFO - Updated partition assignment: [TopicPartition(topic='logs', partition=0)]
2025-09-26 16:00:28,863 - kafka.coordinator.consumer - INFO - Setting newly assigned partitions {TopicPartition(topic='logs', partition=0)} for group anomaly_engine_1758888028
2025-09-26 16:00:28,866 - kafka.consumer.fetcher - INFO - Resetting offset for partition TopicPartition(topic='logs', partition=0) to offset 236035.
2025-09-26 16:00:31,788 - httpx - INFO - HTTP Request: GET http://localhost:6333/collections "HTTP/1.1 200 OK"
2025-09-26 16:00:43,704 - __main__ - INFO - üóëÔ∏è Cleared 1 cached results
2025-09-26 16:01:00,161 - __main__ - INFO - üöÄ Starting Real-Time HDFS Anomaly Detection Engine...
2025-09-26 16:01:00,162 - __main__ - INFO -  Database initialized: anomaly_detection.db
2025-09-26 16:01:00,167 - __main__ - INFO - ‚úÖ Connected to Redis at localhost:6379
2025-09-26 16:01:00,193 - httpx - INFO - HTTP Request: GET http://localhost:6333 "HTTP/1.1 200 OK"
2025-09-26 16:01:00,193 - __main__ - INFO - ‚úÖ Connected to Qdrant at localhost:6333
2025-09-26 16:01:00,193 - __main__ - INFO - üì• Loading ensemble model from ./ensemble/models/line_level_ensemble_v2/line_level_ensemble_v2_results.joblib
2025-09-26 16:01:00,763 - __main__ - INFO - ‚úÖ Loaded ensemble with 3 models
2025-09-26 16:01:00,763 - __main__ - INFO - üéØ Model weights for weighted voting:
2025-09-26 16:01:00,763 - __main__ - INFO -    dt: 0.0790
2025-09-26 16:01:00,763 - __main__ - INFO -    mlp: 0.4315
2025-09-26 16:01:00,763 - __main__ - INFO -    sgd: 0.0747
2025-09-26 16:01:00,763 - __main__ - INFO -    qdrant_similarity: 0.4148
2025-09-26 16:01:00,763 - __main__ - INFO - üì° Enhanced Kafka consumer started
2025-09-26 16:01:00,763 - __main__ - INFO - ‚úÖ Real-Time Anomaly Detection Engine ready!
2025-09-26 16:01:00,764 - kafka.conn - INFO - <BrokerConnection client_id=kafka-python-2.2.15, node_id=bootstrap-0 host=localhost:9092 <connecting> [IPv6 ('::1', 9092, 0, 0)]>: connecting to localhost:9092 [('::1', 9092, 0, 0) IPv6]
2025-09-26 16:01:00,770 - kafka.conn - INFO - <BrokerConnection client_id=kafka-python-2.2.15, node_id=bootstrap-0 host=localhost:9092 <checking_api_versions_recv> [IPv6 ('::1', 9092, 0, 0)]>: Broker version identified as 2.6
2025-09-26 16:01:00,770 - kafka.conn - INFO - <BrokerConnection client_id=kafka-python-2.2.15, node_id=bootstrap-0 host=localhost:9092 <connected> [IPv6 ('::1', 9092, 0, 0)]>: Connection complete.
2025-09-26 16:01:00,770 - kafka.consumer.subscription_state - INFO - Updating subscribed topics to: ('logs',)
2025-09-26 16:01:00,771 - __main__ - INFO - üì° Enhanced Kafka consumer listening to topic: logs
2025-09-26 16:01:00,774 - kafka.conn - INFO - <BrokerConnection client_id=kafka-python-2.2.15, node_id=1 host=localhost:9092 <connecting> [IPv6 ('::1', 9092, 0, 0)]>: connecting to localhost:9092 [('::1', 9092, 0, 0) IPv6]
2025-09-26 16:01:00,879 - kafka.conn - INFO - <BrokerConnection client_id=kafka-python-2.2.15, node_id=1 host=localhost:9092 <connected> [IPv6 ('::1', 9092, 0, 0)]>: Connection complete.
2025-09-26 16:01:00,879 - kafka.conn - INFO - <BrokerConnection client_id=kafka-python-2.2.15, node_id=bootstrap-0 host=localhost:9092 <connected> [IPv6 ('::1', 9092, 0, 0)]>: Closing connection. 
2025-09-26 16:01:00,992 - kafka.cluster - INFO - Coordinator for group/anomaly_engine_1758888060 is BrokerMetadata(nodeId='coordinator-1', host='localhost', port=9092, rack=None)
2025-09-26 16:01:00,992 - kafka.coordinator - INFO - Discovered coordinator coordinator-1 for group anomaly_engine_1758888060
2025-09-26 16:01:00,992 - kafka.coordinator.heartbeat - INFO - Starting new heartbeat thread
2025-09-26 16:01:00,993 - kafka.coordinator.consumer - INFO - Revoking previously assigned partitions set() for group anomaly_engine_1758888060
2025-09-26 16:01:00,993 - kafka.coordinator - INFO - Failed to join group anomaly_engine_1758888060: NodeNotReadyError: coordinator-1
2025-09-26 16:01:00,994 - kafka.conn - INFO - <BrokerConnection client_id=kafka-python-2.2.15, node_id=coordinator-1 host=localhost:9092 <connecting> [IPv6 ('::1', 9092, 0, 0)]>: connecting to localhost:9092 [('::1', 9092, 0, 0) IPv6]
2025-09-26 16:01:01,094 - kafka.coordinator - INFO - Failed to join group anomaly_engine_1758888060: NodeNotReadyError: coordinator-1
2025-09-26 16:01:01,095 - kafka.conn - INFO - <BrokerConnection client_id=kafka-python-2.2.15, node_id=coordinator-1 host=localhost:9092 <connected> [IPv6 ('::1', 9092, 0, 0)]>: Connection complete.
2025-09-26 16:01:01,198 - kafka.coordinator - INFO - (Re-)joining group anomaly_engine_1758888060
2025-09-26 16:01:01,200 - kafka.coordinator - INFO - Received member id kafka-python-2.2.15-b581ba73-3031-47e0-afac-69cf91820c81 for group anomaly_engine_1758888060; will retry join-group
2025-09-26 16:01:01,200 - kafka.coordinator - INFO - Failed to join group anomaly_engine_1758888060: [Error 79] MemberIdRequiredError
2025-09-26 16:01:01,200 - kafka.coordinator - INFO - (Re-)joining group anomaly_engine_1758888060
2025-09-26 16:01:01,212 - kafka.coordinator - INFO - Successfully joined group anomaly_engine_1758888060 <Generation 1 (member_id: kafka-python-2.2.15-b581ba73-3031-47e0-afac-69cf91820c81, protocol: range)>
2025-09-26 16:01:01,213 - kafka.coordinator - INFO - Elected group leader -- performing partition assignments using range
2025-09-26 16:01:01,221 - kafka.consumer.subscription_state - INFO - Updated partition assignment: [TopicPartition(topic='logs', partition=0)]
2025-09-26 16:01:01,221 - kafka.coordinator.consumer - INFO - Setting newly assigned partitions {TopicPartition(topic='logs', partition=0)} for group anomaly_engine_1758888060
2025-09-26 16:01:01,226 - kafka.consumer.fetcher - INFO - Resetting offset for partition TopicPartition(topic='logs', partition=0) to offset 236035.
2025-09-26 16:01:04,397 - httpx - INFO - HTTP Request: GET http://localhost:6333/collections "HTTP/1.1 200 OK"
2025-09-26 16:01:42,703 - __main__ - INFO - üöÄ Starting Real-Time HDFS Anomaly Detection Engine...
2025-09-26 16:01:42,704 - __main__ - INFO -  Database initialized: anomaly_detection.db
2025-09-26 16:01:42,706 - __main__ - INFO - ‚úÖ Connected to Redis at localhost:6379
2025-09-26 16:01:42,732 - httpx - INFO - HTTP Request: GET http://localhost:6333 "HTTP/1.1 200 OK"
2025-09-26 16:01:42,733 - __main__ - INFO - ‚úÖ Connected to Qdrant at localhost:6333
2025-09-26 16:01:42,733 - __main__ - INFO - üì• Loading ensemble model from ./ensemble/models/line_level_ensemble_v2/line_level_ensemble_v2_results.joblib
2025-09-26 16:01:43,266 - __main__ - INFO - ‚úÖ Loaded ensemble with 3 models
2025-09-26 16:01:43,266 - __main__ - INFO - üéØ Model weights for weighted voting:
2025-09-26 16:01:43,266 - __main__ - INFO -    dt: 0.0790
2025-09-26 16:01:43,266 - __main__ - INFO -    mlp: 0.4315
2025-09-26 16:01:43,266 - __main__ - INFO -    sgd: 0.0747
2025-09-26 16:01:43,266 - __main__ - INFO -    qdrant_similarity: 0.4148
2025-09-26 16:01:43,266 - __main__ - INFO - üì° Enhanced Kafka consumer started
2025-09-26 16:01:43,267 - __main__ - INFO - ‚úÖ Real-Time Anomaly Detection Engine ready!
2025-09-26 16:01:43,267 - kafka.conn - INFO - <BrokerConnection client_id=kafka-python-2.2.15, node_id=bootstrap-0 host=localhost:9092 <connecting> [IPv6 ('::1', 9092, 0, 0)]>: connecting to localhost:9092 [('::1', 9092, 0, 0) IPv6]
2025-09-26 16:01:43,275 - kafka.conn - INFO - <BrokerConnection client_id=kafka-python-2.2.15, node_id=bootstrap-0 host=localhost:9092 <checking_api_versions_recv> [IPv6 ('::1', 9092, 0, 0)]>: Broker version identified as 2.6
2025-09-26 16:01:43,275 - kafka.conn - INFO - <BrokerConnection client_id=kafka-python-2.2.15, node_id=bootstrap-0 host=localhost:9092 <connected> [IPv6 ('::1', 9092, 0, 0)]>: Connection complete.
2025-09-26 16:01:43,275 - kafka.consumer.subscription_state - INFO - Updating subscribed topics to: ('logs',)
2025-09-26 16:01:43,275 - __main__ - INFO - üì° Enhanced Kafka consumer listening to topic: logs
2025-09-26 16:01:43,278 - kafka.conn - INFO - <BrokerConnection client_id=kafka-python-2.2.15, node_id=1 host=localhost:9092 <connecting> [IPv6 ('::1', 9092, 0, 0)]>: connecting to localhost:9092 [('::1', 9092, 0, 0) IPv6]
2025-09-26 16:01:43,383 - kafka.conn - INFO - <BrokerConnection client_id=kafka-python-2.2.15, node_id=1 host=localhost:9092 <connected> [IPv6 ('::1', 9092, 0, 0)]>: Connection complete.
2025-09-26 16:01:43,383 - kafka.conn - INFO - <BrokerConnection client_id=kafka-python-2.2.15, node_id=bootstrap-0 host=localhost:9092 <connected> [IPv6 ('::1', 9092, 0, 0)]>: Closing connection. 
2025-09-26 16:01:43,493 - kafka.cluster - INFO - Coordinator for group/anomaly_engine_1758888103 is BrokerMetadata(nodeId='coordinator-1', host='localhost', port=9092, rack=None)
2025-09-26 16:01:43,494 - kafka.coordinator - INFO - Discovered coordinator coordinator-1 for group anomaly_engine_1758888103
2025-09-26 16:01:43,494 - kafka.coordinator.heartbeat - INFO - Starting new heartbeat thread
2025-09-26 16:01:43,495 - kafka.coordinator.consumer - INFO - Revoking previously assigned partitions set() for group anomaly_engine_1758888103
2025-09-26 16:01:43,495 - kafka.coordinator - INFO - Failed to join group anomaly_engine_1758888103: NodeNotReadyError: coordinator-1
2025-09-26 16:01:43,496 - kafka.conn - INFO - <BrokerConnection client_id=kafka-python-2.2.15, node_id=coordinator-1 host=localhost:9092 <connecting> [IPv6 ('::1', 9092, 0, 0)]>: connecting to localhost:9092 [('::1', 9092, 0, 0) IPv6]
2025-09-26 16:01:43,602 - kafka.coordinator - INFO - Failed to join group anomaly_engine_1758888103: NodeNotReadyError: coordinator-1
2025-09-26 16:01:43,603 - kafka.conn - INFO - <BrokerConnection client_id=kafka-python-2.2.15, node_id=coordinator-1 host=localhost:9092 <connected> [IPv6 ('::1', 9092, 0, 0)]>: Connection complete.
2025-09-26 16:01:43,709 - kafka.coordinator - INFO - (Re-)joining group anomaly_engine_1758888103
2025-09-26 16:01:43,717 - kafka.coordinator - INFO - Received member id kafka-python-2.2.15-c371ae3f-69a2-4c44-825c-35b9d08a1ad1 for group anomaly_engine_1758888103; will retry join-group
2025-09-26 16:01:43,717 - kafka.coordinator - INFO - Failed to join group anomaly_engine_1758888103: [Error 79] MemberIdRequiredError
2025-09-26 16:01:43,717 - kafka.coordinator - INFO - (Re-)joining group anomaly_engine_1758888103
2025-09-26 16:01:43,740 - kafka.coordinator - INFO - Successfully joined group anomaly_engine_1758888103 <Generation 1 (member_id: kafka-python-2.2.15-c371ae3f-69a2-4c44-825c-35b9d08a1ad1, protocol: range)>
2025-09-26 16:01:43,740 - kafka.coordinator - INFO - Elected group leader -- performing partition assignments using range
2025-09-26 16:01:43,751 - kafka.consumer.subscription_state - INFO - Updated partition assignment: [TopicPartition(topic='logs', partition=0)]
2025-09-26 16:01:43,751 - kafka.coordinator.consumer - INFO - Setting newly assigned partitions {TopicPartition(topic='logs', partition=0)} for group anomaly_engine_1758888103
2025-09-26 16:01:43,757 - kafka.consumer.fetcher - INFO - Resetting offset for partition TopicPartition(topic='logs', partition=0) to offset 236035.
2025-09-26 16:01:46,992 - httpx - INFO - HTTP Request: GET http://localhost:6333/collections "HTTP/1.1 200 OK"
2025-09-26 16:01:57,189 - __main__ - INFO - üöÄ Starting Real-Time HDFS Anomaly Detection Engine...
2025-09-26 16:01:57,190 - __main__ - INFO -  Database initialized: anomaly_detection.db
2025-09-26 16:01:57,195 - __main__ - INFO - ‚úÖ Connected to Redis at localhost:6379
2025-09-26 16:01:57,222 - httpx - INFO - HTTP Request: GET http://localhost:6333 "HTTP/1.1 200 OK"
2025-09-26 16:01:57,222 - __main__ - INFO - ‚úÖ Connected to Qdrant at localhost:6333
2025-09-26 16:01:57,222 - __main__ - INFO - üì• Loading ensemble model from ./ensemble/models/line_level_ensemble_v2/line_level_ensemble_v2_results.joblib
2025-09-26 16:01:57,775 - __main__ - INFO - ‚úÖ Loaded ensemble with 3 models
2025-09-26 16:01:57,775 - __main__ - INFO - üéØ Model weights for weighted voting:
2025-09-26 16:01:57,775 - __main__ - INFO -    dt: 0.0790
2025-09-26 16:01:57,775 - __main__ - INFO -    mlp: 0.4315
2025-09-26 16:01:57,775 - __main__ - INFO -    sgd: 0.0747
2025-09-26 16:01:57,775 - __main__ - INFO -    qdrant_similarity: 0.4148
2025-09-26 16:01:57,775 - __main__ - INFO - üì° Enhanced Kafka consumer started
2025-09-26 16:01:57,776 - __main__ - INFO - ‚úÖ Real-Time Anomaly Detection Engine ready!
2025-09-26 16:01:57,776 - kafka.conn - INFO - <BrokerConnection client_id=kafka-python-2.2.15, node_id=bootstrap-0 host=localhost:9092 <connecting> [IPv6 ('::1', 9092, 0, 0)]>: connecting to localhost:9092 [('::1', 9092, 0, 0) IPv6]
2025-09-26 16:01:57,782 - kafka.conn - INFO - <BrokerConnection client_id=kafka-python-2.2.15, node_id=bootstrap-0 host=localhost:9092 <checking_api_versions_recv> [IPv6 ('::1', 9092, 0, 0)]>: Broker version identified as 2.6
2025-09-26 16:01:57,782 - kafka.conn - INFO - <BrokerConnection client_id=kafka-python-2.2.15, node_id=bootstrap-0 host=localhost:9092 <connected> [IPv6 ('::1', 9092, 0, 0)]>: Connection complete.
2025-09-26 16:01:57,783 - kafka.consumer.subscription_state - INFO - Updating subscribed topics to: ('logs',)
2025-09-26 16:01:57,783 - __main__ - INFO - üì° Enhanced Kafka consumer listening to topic: logs
2025-09-26 16:01:57,786 - kafka.conn - INFO - <BrokerConnection client_id=kafka-python-2.2.15, node_id=1 host=localhost:9092 <connecting> [IPv6 ('::1', 9092, 0, 0)]>: connecting to localhost:9092 [('::1', 9092, 0, 0) IPv6]
2025-09-26 16:01:57,891 - kafka.conn - INFO - <BrokerConnection client_id=kafka-python-2.2.15, node_id=1 host=localhost:9092 <connected> [IPv6 ('::1', 9092, 0, 0)]>: Connection complete.
2025-09-26 16:01:57,891 - kafka.conn - INFO - <BrokerConnection client_id=kafka-python-2.2.15, node_id=bootstrap-0 host=localhost:9092 <connected> [IPv6 ('::1', 9092, 0, 0)]>: Closing connection. 
2025-09-26 16:01:58,003 - kafka.cluster - INFO - Coordinator for group/anomaly_engine_1758888117 is BrokerMetadata(nodeId='coordinator-1', host='localhost', port=9092, rack=None)
2025-09-26 16:01:58,003 - kafka.coordinator - INFO - Discovered coordinator coordinator-1 for group anomaly_engine_1758888117
2025-09-26 16:01:58,004 - kafka.coordinator.heartbeat - INFO - Starting new heartbeat thread
2025-09-26 16:01:58,004 - kafka.coordinator.consumer - INFO - Revoking previously assigned partitions set() for group anomaly_engine_1758888117
2025-09-26 16:01:58,004 - kafka.coordinator - INFO - Failed to join group anomaly_engine_1758888117: NodeNotReadyError: coordinator-1
2025-09-26 16:01:58,006 - kafka.conn - INFO - <BrokerConnection client_id=kafka-python-2.2.15, node_id=coordinator-1 host=localhost:9092 <connecting> [IPv6 ('::1', 9092, 0, 0)]>: connecting to localhost:9092 [('::1', 9092, 0, 0) IPv6]
2025-09-26 16:01:58,108 - kafka.coordinator - INFO - Failed to join group anomaly_engine_1758888117: NodeNotReadyError: coordinator-1
2025-09-26 16:01:58,109 - kafka.conn - INFO - <BrokerConnection client_id=kafka-python-2.2.15, node_id=coordinator-1 host=localhost:9092 <connected> [IPv6 ('::1', 9092, 0, 0)]>: Connection complete.
2025-09-26 16:01:58,214 - kafka.coordinator - INFO - (Re-)joining group anomaly_engine_1758888117
2025-09-26 16:01:58,222 - kafka.coordinator - INFO - Received member id kafka-python-2.2.15-25dc5f55-6392-468c-bcb3-06d890de4027 for group anomaly_engine_1758888117; will retry join-group
2025-09-26 16:01:58,223 - kafka.coordinator - INFO - Failed to join group anomaly_engine_1758888117: [Error 79] MemberIdRequiredError
2025-09-26 16:01:58,223 - kafka.coordinator - INFO - (Re-)joining group anomaly_engine_1758888117
2025-09-26 16:01:58,245 - kafka.coordinator - INFO - Successfully joined group anomaly_engine_1758888117 <Generation 1 (member_id: kafka-python-2.2.15-25dc5f55-6392-468c-bcb3-06d890de4027, protocol: range)>
2025-09-26 16:01:58,245 - kafka.coordinator - INFO - Elected group leader -- performing partition assignments using range
2025-09-26 16:01:58,260 - kafka.consumer.subscription_state - INFO - Updated partition assignment: [TopicPartition(topic='logs', partition=0)]
2025-09-26 16:01:58,261 - kafka.coordinator.consumer - INFO - Setting newly assigned partitions {TopicPartition(topic='logs', partition=0)} for group anomaly_engine_1758888117
2025-09-26 16:01:58,266 - kafka.consumer.fetcher - INFO - Resetting offset for partition TopicPartition(topic='logs', partition=0) to offset 236035.
2025-09-26 16:02:01,530 - httpx - INFO - HTTP Request: GET http://localhost:6333/collections "HTTP/1.1 200 OK"
